src/twisted/internet/udp.py

```diff
@@ -193,7 +193,11 @@ def _bindSocket(self):
             # Create a new socket and make it listen
             try:
                 skt = self.createInternetSocket()
-                skt.bind((self.interface, self.port))
+                if self.addressFamily == socket.AF_INET6:
+                    addr = socket.getaddrinfo(self.interface, self.port, 0, 0, 0, socket.AI_NUMERICHOST )[0][4]
```

Comment:
To fully avoid any I/O, you want to use `AI_NUMERICSERV` as well… i.e. what `_resolveIPv6` does in `tcp.py`.  So I think you want to maybe refactor `_resolveIPv6` into something in maybe `twisted.internet.address` or something, and call it from here rather than calling `getaddrinfo` directly, since `[4]` is also a bit of a silly thing to re-encode here.